core/common/src/main/java/alluxio/master/MasterInquireClient.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.util.network.NetworkAddressUtils.ServiceType;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * Client for determining the primary master.
+ */
+public interface MasterInquireClient {
```

Comment:
This name does not feel very intuitive, but unfortunately I don't have a great suggestion.